app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/Indexer/ModifySelectInProductPriceIndexFilter.php

```diff
@@ -39,22 +41,30 @@ class ModifySelectInProductPriceIndexFilter
      */
     private $stockByWebsiteIdResolver;
 
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
     /**
      * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
      * @param StockConfigurationInterface $stockConfiguration
      * @param ResourceConnection $resourceConnection
      * @param StockByWebsiteIdResolverInterface $stockByWebsiteIdResolver
+     * @param DefaultStockProviderInterface $defaultStockProvider
      */
     public function __construct(
         StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
         StockConfigurationInterface $stockConfiguration,
         ResourceConnection $resourceConnection,
-        StockByWebsiteIdResolverInterface $stockByWebsiteIdResolver
+        StockByWebsiteIdResolverInterface $stockByWebsiteIdResolver,
+        DefaultStockProviderInterface $defaultStockProvider
```

Comment:
Breaking change see 
https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/

Reply:
@larsroettig, Backward Compatibility Policy is not applied to Plugins, Observers and Setup Scripts
See https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/